### DISCUSSION TITLE
How to send file with HttpRequest component?

### DISCUSSION THREAD

**Discussion by martina-parkanova** [linkedpipes/etl#979] [Q&A] [answered]
Could you please help me, how can I send file when using HttpRequest component? I have a file produced from RDF to file component and I tried to use configuration via the text holder as is in the documentation. But I don't know, how to write the information about the file in the configuration, now I am only able to make successful GET requests and not the POST ones. 
![image](https://github.com/linkedpipes/etl/assets/152961774/d0e1cc7f-2f9b-4d9d-b892-5ba30bca4ce1)
In my configuration, I have something like this (but the file part is definitely wrong, I just don't know how to write it there...): 
![image](https://github.com/linkedpipes/etl/assets/152961774/a870bcac-e3d2-416d-b1ff-778d31f32e30)

**Comment by jakubklimek** [comment]
An example of a configuration of sending a file named `configuration.json` via POST to Slack:
```turtle
@prefix req: <http://plugins.linkedpipes.com/ontology/x-httpRequest#> .

<http://localhost/slack> a req:Task ;
  req:url "https://hooks.slack.com/services/<insert_token>" ;
  req:method "POST" ;
  req:postContentAsBody true ;
  req:content <http://localhost/slack/content> .

<http://localhost/slack/content> a req:Content ;
  req:fileReference "configuration.json" .
```

**Reply by martina-parkanova** [reply]
Thanks, I tried to add your solution to mine, but the file is still not sent with the request, according to my application receiving the request, can I see somewhere the actual request that was sent from linkedpipes? Because network inspect only shows call to linkedpipes executor, and executor doesn't have anything in docker logs... 

![image](https://github.com/linkedpipes/etl/assets/152961774/09a6175f-ae98-40aa-b01e-1f0a85a20ef5)

**Comment by jakubklimek** [comment]
Is the file really called `data.xml`? In your screenshot, you pass an output of RDF to File - and `.xml` is not a usual file extension for RDF serializations...

You can check the logs of the execution itself.

Also, I can see that your `Header` instance is wrong - if you try to pass two headers, you need two `Header` instances, each with a "name/value" pair.

**Reply by martina-parkanova** [reply]
I tried to rewrite the configuration... 
![image](https://github.com/linkedpipes/etl/assets/152961774/638dd6b8-ff7d-49c3-9a93-d080c02e4064)

I also looked at the logs from the execution, I don't see the actual request there, but I can see logs telling:
2024-02-08 10:57:03 09:57:03,156 [pool-18-thread-1] DEBUG com.linkedpipes.etl.executor.api.v1.component.task.TaskConsumerWrap - Starting task 'http://cca.cz'
2024-02-08 10:57:03 09:57:03,156 [pool-18-thread-1] INFO  com.linkedpipes.plugin.http.request.main.TaskExecutor        - Executing 'POST' on 'http://mynetwork:8080/uploadFile to 'null'

And yes, I know that I shouldn't use rdf/xml, but I have it like that right now, so it really outputs file called data.xml:
![image](https://github.com/linkedpipes/etl/assets/152961774/cd328788-7b7f-420a-97bc-86a330d70780)

**Comment by jakubklimek** [comment, marked as answer]
See [this execution](https://demo.etl.linkedpipes.com/#/pipelines/edit/canvas?pipeline=https:%2F%2Fdemo.etl.linkedpipes.com%2Fresources%2Fpipelines%2F1596028447599&execution=https:%2F%2Fdemo.etl.linkedpipes.com%2Fresources%2Fexecutions%2F1707387272928-2196-f632eb58-dd48-4148-954d-abec9aa6e4dc) for another working example. Maybe the issue is with the `multipart/form-data` header?

**Reply by martina-parkanova** [reply]
Yes, that was the problem - when I tried to use only application/octet-stream, everything worked. And I can live with that type of transfer, so thank you very much!
